### DISCUSSION TITLE
Request-PnPPersonalSite

### DISCUSSION THREAD

**Discussion by Karspi** [pnp/powershell#528] [Q&A]
Hello,

we are trying to automate a migration with PowerShell and we are stuck atm by the Request-PnPPersonalSite
The similar command from SPO PS Module - Request-SPOPersonalSite works perfeclty, unfortunately SPO Module does not allow unattended login therefore we have to use the new module.

The issue that we face is that even with global admin user and admin consent for the permissions of the Enterprise Application PnP Management Shell we receive the error 

```
Request-PnPPersonalSite : Access denied. You do not have permission to perform this action or access this resource.
At line:1 char:1
+ Request-PnPPersonalSite -UserEmails ****
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Request-PnPPersonalSite], ServerUnauthorizedAccessException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Admin.RequestPersonalSite
```

What we miss here? Thank you very much

**Comment by Deweyoxberg** [comment]
It's hard to tell what user the connection attempt is executing under. First thing I would check is that the account you're using to connect has the required permissions to start with, namely:

- Sharepoint Administrator
- Licensed for Sharepoint Online

[Request-PNPPersonalSite Documentation](https://pnp.github.io/powershell/cmdlets/Request-PnPPersonalSite.html)

If that has been validated, then I did find [#2267 - New-PnPPersonalSite - Access Denied using Azure AD App Identity](https://github.com/pnp/PnP-PowerShell/issues/2267). That post lead me to [#4673](https://github.com/SharePoint/sp-dev-docs/issues/4673). From #4673 I found [this FAQ](https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azuread#faq):

> ### **Can I use other means besides certificates for realizing app-only access for my Azure AD app?**
> No, all other options are blocked by SharePoint Online and will result in an Access Denied message.

The same person who posted the above FAQ also [posted a workaround](https://github.com/SharePoint/sp-dev-docs/issues/4673#issuecomment-642046096) that worked for them. However, it's unclear how they're connecting to their tenant though I assume it is certificate-based.

**Comment by BIERAAP** [comment]
anyone managed to get it working?
